app/guid-file/controller.ts

```diff
@@ -180,7 +193,131 @@ export default class GuidFile extends Controller {
 
     @action
     versionChange(version: number) {
-        this.set('revision', +version);
+        this.set('revision', version);
+    }
+
+    @action
+    async toggleVersions() {
+        this.toggleProperty('revisionClicked');
+
+        if (this.revisionClicked === true) {
+            this.openVersions();
+        }
+
+        if (this.tagsClicked === true) {
+            this.closeTags();
+        }
+
+        if (this.revisionClicked === false) {
+            this.closeVersions();
+        }
+    }
+
+    @action
+    async toggleTags() {
+        this.toggleProperty('tagsClicked');
+
+        if (this.tagsClicked === true) {
+            this.openTags();
+        }
+        if (this.revisionClicked === true) {
+            this.closeVersions();
+        }
+        if (this.tagsClicked === false) {
+            this.closeTags();
+        }
+    }
+
+    @action
+    async showMoreButtons() {
+        console.log('Open more button clicked');
+    }
+
+    @action
+    async share() {
+        console.log('File shared with service:');
+    }
+
+    openVersions() {
+
+        if (this.tagsClicked === true) {
+            this.closeTags();
+            this.tagsClicked = false;
+        }
+
+        const rightPanel = document.getElementById('rightPanel');
+        if (rightPanel) {
+            rightPanel.style.marginLeft = '-400px';
+            rightPanel.classList.replace('col-sm-1', 'col-sm-5');
+        }
+
+        const versionSlide = document.getElementById('versions');
+        if (versionSlide) {
+            // versionSlide.removeAttribute('hidden');
+            versionSlide.hidden = false;
+            versionSlide.style.width = '400px';
+        }
+    }
+
+    closeVersions() {
+        const versionSlide = document.getElementById('versions');
+        if (versionSlide) {
+            versionSlide.hidden = true;
+            versionSlide.style.width = '0px';
+            // versionSlide.addAttribute('hidden');
+        }
+
+        const rightPanel = document.getElementById('rightPanel');
+        if (rightPanel) {
+            rightPanel.style.marginLeft = '0px';
+            rightPanel.classList.replace('col-sm-5', 'col-sm-1');
+        }
+    }
```

Comment:
These are not necessary.

Reply:
The rightPanel logic and comments have since been removed, marking this as resolved for now.